sdcm/utils/toppartition_util.py

```diff
@@ -0,0 +1,143 @@
+import random
+import logging
+import re
+
+from typing import List, Tuple
+from abc import abstractmethod, ABC
+from collections import OrderedDict
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TopPartitionCmd(ABC):
+
+    def __init__(self, ks_cf_list: dict):
+        self.ks_cf_list: list = ks_cf_list
+        self._built_args: dict = {}
+
+    @staticmethod
+    def _parse_toppartitions_output(output: str) -> dict:
+        """parsing output of toppartitions
+
+        input format stored in output parameter:
+        WRITES Sampler:
+          Cardinality: ~10 (15 capacity)
+          Top 10 partitions:
+            Partition     Count       +/-
+            (ks0:cf3) 9        11         0
+            (ks0:cf3) 0         1         0
+            (ks0:cf3) 1         1         0
+
+        READS Sampler:
+          Cardinality: ~10 (256 capacity)
+          Top 3 partitions:
+            Partition     Count       +/-
+            (ks0:cf3) 0         3         0
+            (ks0:cf3) 1         3         0
+            (ks0:cf3) 2         3         0
+            (ks0:cf3) 3         2         0
+
+        return Dict:
+        {
+            'READS': {
+                'toppartitions': '10',
+                'partitions': OrderedDict('(ks0:cf3) 0': {'count': '1', 'margin': '0'},
+                                          '(ks0:cf3) 1': {'count': '1', 'margin': '0'},
+                                          '(ks0:cf3) 2': {'count': '1', 'margin': '0'}),
+                'cardinality': '10',
+                'capacity': '256',
+            },
+            'WRITES': {
+                'toppartitions': '10',
+                'partitions': OrderedDict('(ks0:cf3) 10': {'count': '1', 'margin': '0'},
+                                          '(ks0:cf3) 11': {'count': '1', 'margin': '0'},
+                                          '(ks0:cf3) 21': {'count': '1', 'margin': '0'}),
+                'cardinality': '10',
+                'capacity': '256',
+                'sampler': 'WRITES'
+            }
+        }
+
+
+        Arguments:
+            output {str} -- stdout of nodetool topparitions command
+
+        Returns:
+            dict -- result of parsing
+        """
+        pattern1 = r"(?P<sampler>[A-Z]+)\sSampler:\W+Cardinality:\s~(?P<cardinality>[0-9]+)\s\((?P<capacity>[0-9]+)\scapacity\)\W+Top\s(?P<toppartitions>[0-9]+)\spartitions:"  # pylint: disable=line-too-long
+        pattern2 = r"(?P<partition>\([\w:]+\)\s[\w:]+)\s+(?P<count>[\d]+)\s+(?P<margin>[\d]+)"
+        toppartitions = {}
+        for out in output.strip().split('\n\n'):
+            partition = OrderedDict()
+            sampler_data = re.search(pattern1, out, re.MULTILINE)
+            assert sampler_data, f"Pattern:{pattern1} are not matched on string:\n {out}"
+            sampler_data = sampler_data.groupdict()
+            partitions = re.findall(pattern2, out, re.MULTILINE)
```

Comment:
if we don't expect partitions (empty result) then it will passed. If partitions are expected but they are not found by regexp,
assert exception will be raised in verify_output method